src/components/hmi_message_handler/src/websocket_session.cc

```diff
@@ -304,12 +304,13 @@ void WebsocketSession::LoopThreadDelegate::exitThreadMain() {
 }
 
 void WebsocketSession::LoopThreadDelegate::DrainQueue() {
-  while (!message_queue_.empty()) {
-    Message message_ptr;
-    message_queue_.pop(message_ptr);
-    if (!shutdown_) {
-      handler_.ws_.write(boost::asio::buffer(*message_ptr));
-    };
+  Message message_ptr;
+  while (message_queue_.pop(message_ptr) && !shutdown_) {
+    boost::system::error_code ec;
+    handler_.ws_.write(boost::asio::buffer(*message_ptr), ec);
+    if (ec) {
+      LOG4CXX_ERROR(ws_logger_, "A system error has occurred: " << ec);
```

Comment:
@SNiukalov Add detail about the error please

Reply:
@AByzhynar 
Thanks, fixed in: https://github.com/smartdevicelink/sdl_core/pull/3191/commits/ccc2795f8cdbaf2f543d92b38f2183831538b3c9